app/src/main/java/com/sopt/now/compose/feature/auth/SignUpActivity.kt

```diff
@@ -13,153 +16,77 @@ import androidx.compose.material3.MaterialTheme
-
-        return validationMessages
-    }
-
-    private fun checkIdLength(
-        user: User,
-        validationMessages: MutableList<String>
-    ) {
-        validateLength(user.id, MIN_ID_LENGTH, MAX_ID_LENGTH, R.string.login_id_text)?.let {
-            validationMessages.add(it)
-        }
-    }
-
-    private fun checkPwdLength(
-        user: User,
-        validationMessages: MutableList<String>
-    ) {
-        validateLength(
-            user.password,
-            MIN_PWD_LENGTH,
-            MAX_PWD_LENGTH,
-            R.string.login_password_text
-        )?.let {
-            validationMessages.add(it)
-        }
-    }
-
-    private fun checkMbtiLength(
-        user: User,
-        validationMessages: MutableList<String>
-    ) {
-        if (user.mbti.length != MBTI_LENGTH) {
-            validationMessages.add(getString(R.string.sign_up_mbti_snack_message, MBTI_LENGTH))
-        }
-    }
-
-    private fun checkNickNameLength(
-        user: User,
-        validationMessages: MutableList<String>
-    ) {
-        if (user.nickName.length <= MIN_NICKNAME_LENGTH) {
-            validationMessages.add(
-                getString(
-                    R.string.sign_up_nickname_snack_message,
-                    MIN_NICKNAME_LENGTH
-                )
-            )
-        }
-    }
-
-    private fun validateLength(
-        input: String,
-        minLength: Int,
-        maxLength: Int,
-        fieldNameResId: Int
-    ): String? {
-        return if (input.length !in minLength..maxLength) {
-            getString(
-                R.string.sign_up_snack_message,
-                getString(fieldNameResId),
-                minLength,
-                maxLength
-            )
-        } else {
-            null
-        }
-    }
-
-    private fun handleValidationResult(validationMessages: List<String>, userInputData: User) {
-        if (validationMessages.isEmpty()) {
-            navigateToLoginActivity(userInputData)
-            toast("회원가입이 완료되었습니다.")
-        } else {
-            toast(validationMessages.joinToString("\n"))
-        }
-    }
-
-    private fun navigateToLoginActivity(userInputData: User) {
-        Intent().apply {
-            putExtra(KeyStorage.USER_INPUT, userInputData)
-            setResult(RESULT_OK, this)
-            finish()
-        }
-    }
-
-    companion object {
-        const val MIN_ID_LENGTH = 6
-        const val MAX_ID_LENGTH = 10
-        const val MIN_PWD_LENGTH = 8
-        const val MAX_PWD_LENGTH = 12
-        const val MIN_NICKNAME_LENGTH = 2
-        const val MBTI_LENGTH = 4
-    }
 }
 
 @Composable
 fun SignUpScreen(
-    onSignUpBtnClick: (User) -> Unit
+    onSignUpBtnClick: (User) -> Unit,
+    viewModel: SignUpViewModel
 ) {
     var id by remember { mutableStateOf("") }
     var password by remember { mutableStateOf("") }
     var nickName by remember { mutableStateOf("") }
     var mbti by remember { mutableStateOf("") }
```

Comment:
[P1]
rememberSaveable을 사용하면 구성변경 시에도 데이터를 보존할 수 있습니다.
```suggestion
    var id by rememberSaveable { mutableStateOf("") }
    var password by rememberSaveable { mutableStateOf("") }
    var nickName by rememberSaveable { mutableStateOf("") }
    var mbti by rememberSaveable { mutableStateOf("") }
```

Reply:
헉 진짜 꿀팁이네요!!

Reply:
[985415d](https://github.com/NOW-SOPT-ANDROID/chanwoo-bae/pull/17/commits/985415d1fa47c25298b5461f17e26601d86ed81e) 수정했습니다!